core/services/fluxmonitor/flux_monitor.go

```diff
@@ -56,12 +56,15 @@ type Service interface {
 	RemoveJob(models.JobID)
 	Start() error
 	Stop()
+	SetLogger(logger *logger.Logger)
 }
 
 type concreteFluxMonitor struct {
+	mut            sync.Mutex
```

Comment:
Change the name to `muLogger` or `loggerMu` or something so that it's clear exactly what's being mutexed (we wouldn't want to share this mutex across multiple concerns in `concreteFluxMonitor`)